src/reconcile/rec_visibility.c

```diff
@@ -633,6 +631,13 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         upd_saved = upd_select->upd_saved = true;
     }
 
+    /*
+     * Set statistics for update restore evictions. Update restore eviction debug mode forces update
+     * restores to both committed or uncommitted changes.
+     */
+    if (supd_restore || F_ISSET(r, WT_REC_SCRUB))
```

Comment:
This will affect the behaviour of actual scrubbing. Is that what we want?

Reply:
Earlier with SCRUB, we are performing update restore, but that is not considered in any of the stats. With the above change,  all the SCRUB evictions are considered as update restore evictions.

Reply:
Right. Well, somewhere in the code we have a check that does update restore if we're scrubbing. Can that be removed since we're using `r->cache_write_restore` in both cases?

Reply:
The problem here is, the stats are updated based on `r->cache_write_restore` and the actual update restore is performed based on `SCRUB` flag or `multi->supd_restore` flag later. We need to combine the stats and update restore into a single place.

Reply:
Yeah, I see that check. But isn't `multi->supd_restore` set based on `r->cache_write_restore`? Why do we still need to check for the `SCRUB` flag after this point? If the `SCRUB` is being checked here, then I'd expect the `SCRUB` check in `rec_write` to be removed.

If it's hard to refactor this, then fine. But there's clearly some redundancy going on here.

Reply:
I agree there is a little redundancy, will make an effort to remove this.

Reply:
Sounds good. If it's non-trivial, feel free to just merge this. We can always clean this up later down the track.

Reply:
Attempted this. Looks like its going to be a little be harder than anticipated. Planning to merge this